### PR TITLE
Make sure `ParamSpec` suffix and arg kind do match

### LINKED ISSUE #13460: `ParamSpec`: mypy is happy for args to be annotated with `P.kwargs` and kwargs to be annotated with `P.args`
**Bug Report**

Mypy has no issues with this snippet of code. It probably should:

```python
from typing import Callable
from typing_extensions import ParamSpec

P = ParamSpec("P")

def func(callback: Callable[P, str]) -> Callable[P, str]:
    def inner(*args: P.kwargs, **kwargs: P.args) -> str:
        return 'baz'
    return inner
```

This bug was accidentally discovered in #13459.

**To Reproduce**

https://mypy-play.net/?mypy=latest&python=3.10&gist=87f5f5277409f0fd77675e8dcc6a11fc

**Expected Behavior**

Mypy should emit two errors for this snippet:

1. `*args` have been annotated with `P.kwargs` instead of `P.args`
2. `**kwargs` have been annotated with `P.args` instead of `P.kwargs`

**Actual Behavior**

No error is emitted.

### PR DESCRIPTION
**Pull request by sobolevn**
I am mostly worried about the error message here :)

Closes #13460

### CODE REVIEW

File: test-data/unit/check-parameter-specification.test [hunk 1]
```diff
@@ -1112,3 +1112,18 @@ def func(callback: Callable[P, str]) -> Callable[P, str]:
...
+from typing_extensions import ParamSpec
+
+P = ParamSpec("P")
+
+def func(callback: Callable[P, str]) -> Callable[P, str]:
+    def inner(
+        *args: P.kwargs,   # E: Use ".args" for "*" argument
+        **kwargs: P.args,  # E: Use ".kwargs" for "**" argument
+    ) -> str:
+        return "foo"
```
Comment (AlexWaygood) [inline test-data/unit/check-parameter-specification.test:1127]: Maybe add some `reveal_type`s here, as a sanity check to make sure that the typo doesn't cause mypy to make any incorrect inferences elsewhere?

```suggestion
        reveal_type(args)  # N: Revealed type is "builtins.tuple[builtins.object, ...]"
        reveal_type(kwargs)  # N: Revealed type is "builtins.dict[builtins.str, builtins.object]"
        return "foo"
```

File: mypy/typeanal.py [hunk 2]
```diff
@@ -847,8 +846,12 @@ def anal_star_arg_type(self, t: Type, kind: ArgKind, nested: bool) -> Type:
                 if isinstance(tvar_def, ParamSpecType):
                     if kind == ARG_STAR:
                         make_paramspec = paramspec_args
+                        if components[-1] == "kwargs":
+                            self.fail('Use ".args" for "*" argument', t)
```
Comment (AlexWaygood) [inline mypy/typeanal.py:850]:
```suggestion
                            self.fail(f'Use "{components[:-1]}.args" for variadic "*" parameter', t)
```

File: mypy/typeanal.py [hunk 3]
```diff
@@ -847,8 +846,12 @@ def anal_star_arg_type(self, t: Type, kind: ArgKind, nested: bool) -> Type:
...
                     elif kind == ARG_STAR2:
                         make_paramspec = paramspec_kwargs
+                        if components[-1] == "args":
+                            self.fail('Use ".kwargs" for "**" argument', t)
```
Comment (AlexWaygood) [inline mypy/typeanal.py:854]:
```suggestion
                            self.fail(f'Use "{components[:-1]}.kwargs" for variadic "**" parameter', t)
```

**Review by AlexWaygood — COMMENTED** [review]
Maybe:

File: test-data/unit/check-parameter-specification.test (hunk 1 above)
Comment (sobolevn) [reply test-data/unit/check-parameter-specification.test:1127]: This is not true:

```
main:11: note: Revealed type is "P1.args`-1"  (diff)
main:12: note: Revealed type is "P1.kwargs`-1" (diff)
```

Comment (AlexWaygood) [reply test-data/unit/check-parameter-specification.test:1127]: > This is not true:
> 
> ```
> main:11: note: Revealed type is "P1.args`-1"  (diff)
> main:12: note: Revealed type is "P1.kwargs`-1" (diff)
> ```

Oh, right. Maybe this, in that case?

```suggestion
        a = args[0]
        reveal_type(a)  # N: Revealed type is "builtins.object"
        for b, c in kwargs.items():
            reveal_type(b)  # N: Revealed type is "builtins.str"
            reveal_type(c)  # N: Revealed type is "builtins.object"
        return "foo"
```

Comment (sobolevn) [reply test-data/unit/check-parameter-specification.test:1127]: I think this is too much :)

Comment (AlexWaygood) [reply test-data/unit/check-parameter-specification.test:1127]: Fair enough :)

**Review by AlexWaygood — APPROVED** [review]
LGTM!

**Comment by sobolevn** [comment]
I will leave this open for a couple of days, maybe some other ideas about error message will araise :)

File: mypy/typeanal.py [hunk 4]
```diff
@@ -838,17 +838,21 @@ def anal_type_guard_arg(self, t: UnboundType, fullname: str) -> Type | None:
...
             components = t.name.split(".")
-            sym = self.lookup_qualified(".".join(components[:-1]), t)
+            tvar_name = ".".join(components[:-1])
+            sym = self.lookup_qualified(tvar_name, t)
             if sym is not None and isinstance(sym.node, ParamSpecExpr):
                 tvar_def = self.tvar_scope.get_binding(sym)
                 if isinstance(tvar_def, ParamSpecType):
                     if kind == ARG_STAR:
                         make_paramspec = paramspec_args
+                        if components[-1] == "kwargs":
```
Comment (JelleZijlstra) [inline mypy/typeanal.py:850]:
```suggestion
                        if components[-1] != "args":
```
Just in case some other attribute comes up.

**Review by JelleZijlstra — APPROVED** [review]
Thanks, one nit.

Another weirdness I noticed: if you do `def inner(*args: P.__bound__, **kwargs: P.kwargs):`, you get `main.py:6: error: Name "P.__bound__" is not defined` (expected though the wording could be improved) but also `main.py:6: error: Name "P.kwargs" is not defined` (not expected).

**Comment by sobolevn** [comment]
> Another weirdness I noticed: if you do def inner(*args: P.__bound__, **kwargs: P.kwargs):, you get main.py:6: error: Name "P.__bound__" is not defined (expected though the wording could be improved) but also main.py:6: error: Name "P.kwargs" is not defined (not expected).

Thanks! This is something else. I will take a look later.

Good catch about `!=`, refactored.

**Pull request merged**